src/java.base/share/classes/jdk/internal/foreign/abi/LinkerOptions.java

```diff
@@ -137,18 +138,57 @@ public void validateForDowncall(FunctionDescriptor descriptor) {
                 throw new IllegalArgumentException("Index '" + index + "' not in bounds for descriptor: " + descriptor);
             }
         }
+
+        @Override
+        public int hashCode() {
+            return index;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return obj instanceof FirstVariadicArg that && index == that.index;
```

Comment:
Might be worth checking `obj == this || obj instanceof ...`

Reply:
I think this is still unlikely to hit (it is possible if people intentionally reuse the value of a `Linker.Option.firstVariadicArg` call), most of the usages will see non-identity equality comparisons.